### DISCUSSION TITLE
Second derivative for FTIR spectra

### DISCUSSION THREAD

**Discussion by sashahexe** [spectrochempy/spectrochempy#475] [Ideas]
Hi,

Thank you very much for this very useful package. I use it for FTIR analysis in my work a lot . 
It would be very useful to have a method to automate obtaining a second derivative for batches of IR spectra as it often can be advantageous when monitoring the peaks intensity changes over time. What do you think?

Best regards,
Alexandra

**Comment by atravert** [comment]
Dear Alexandra,

Thank you for your interest.

Derivatives are implemented but not described yet in the userguide. The function to get derivatives is [savgol_filter()](https://www.spectrochempy.fr/latest/userguide/reference/generated/spectrochempy.NDDataset.savgol_filter.html).

It can be used to smooth spectra as documented [here](https://www.spectrochempy.fr/latest/userguide/processing/smoothing.html?highlight=savgol
)

To get the second derivatives, you have to set the 'deriv' - which defaukt is 0 - parameter to 2.

I will document it in a next release. In the meantime, let us know if you need further information !

Best regards
Arnaud

**Reply by sashahexe** [reply]
Dear Arnaud, 

ahha! This is great, thank you for the hint!

Kind regards,
Alexandra

**Reply by fernandezc** [reply]
You can look this as it may be useful for your problem : https://www.spectrochempy.fr/latest/userguide/analysis/peak_integration.html.
Best regards

**Reply by sashahexe** [reply]
Thank you! I’ve realised you already have it, somehow I didn’t spot it before.
